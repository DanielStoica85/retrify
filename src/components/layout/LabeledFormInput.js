import React from 'react';
import { FormGroup, Label, Input } from 'reactstrap';

const LabeledFormInput = props => {
    const {
        labelText,
        labelFor,
        inputType,
        inputName,
        inputId,
        inputPlaceholder,
        handleChange,
        value
    } = props;
    return (
        <FormGroup>
            <Label for={labelFor}>{labelText}</Label>
            <Input
                type={inputType}
                name={inputName}
                id={inputId}
                placeholder={inputPlaceholder}
                onChange={handleChange}
                value={value}
            />
        </FormGroup>
    );
};

export default LabeledFormInput;
