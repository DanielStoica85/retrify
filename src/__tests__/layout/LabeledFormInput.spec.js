import React from 'react';
import renderer from 'react-test-renderer';
import LabeledFormInput from '../../components/layout/LabeledFormInput';

describe('LabeledFormInput component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<LabeledFormInput />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
