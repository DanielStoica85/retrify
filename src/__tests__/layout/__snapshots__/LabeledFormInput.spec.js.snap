// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledFormInput component renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    className=""
  />
  <input
    className="form-control"
    type="text"
  />
</div>
`;
